src/decays/experimental_constraints.cpp

```diff
@@ -0,0 +1,579 @@
+
+   // check location of databases
+   // HiggsBounds
+   if (higgsbounds_dataset.empty()) {
+      throw SetupError("Need to specify location of HiggsBounds database");
+   }
+   else if (!std::filesystem::exists(higgsbounds_dataset)) {
+      throw SetupError("No HiggsBounds database found at " + higgsbounds_dataset);
+   }
+   // HiggsSignals
+   if (higgssignals_dataset.empty()) {
+      throw SetupError("Need to specify location of HiggsSignals database");
+   }
+   else if (!std::filesystem::exists(higgssignals_dataset)) {
+      throw SetupError("No HiggsSignals database found at " + higgssignals_dataset);
+   }
+
+   auto pred = Higgs::Predictions();
+   namespace HP = Higgs::predictions;
+
+   for (auto const& el : bsm_input) {
+      auto effc = HP::NeutralEffectiveCouplings {};
+      auto& s = pred.addParticle(HP::BsmParticle(el.particle, HP::ECharge::neutral, static_cast<HP::CP>(el.CP)));
+      s.setMass(el.mass);
+      s.setMassUnc(relMassError*el.mass); // set mass uncertainty to 3%
+
+      // quarks
+      effc.dd = el.dd.second;
+      effc.uu = el.uu.second;
+      effc.ss = el.ss.second;
+      effc.cc = el.cc.second;
+      effc.bb = el.bb.second;
+      effc.tt = el.tt.second;
+
+      // leptons
+      effc.ee = el.ee.second;
+      effc.mumu = el.mumu.second;
+      effc.tautau = el.tautau.second;
+
+      // gauge bosons
+      effc.WW = el.WW.second;
+      effc.ZZ = el.ZZ.second;
+      effc.gamgam = el.gamgam.second;
+      effc.Zgam = el.Zgam.second;
+      effc.gg = el.gg.second;
+
+      effc.lam = el.lam;
+
+      effectiveCouplingInput(s, effc, refModel, calcggH, calcHgamgam);
+
+      // Effective coupligs below are defined as sqrt(Gamma CP-even) + I sqrt(Gamma CP-odd)
+      // (note that this is different than couplings like gg, WW etc)
+      // so taking a norm gives a total partial width
+      s.setDecayWidth(HP::Decay::emu,   std::norm(el.emu.second));
+      s.setDecayWidth(HP::Decay::etau,  std::norm(el.etau.second));
+      s.setDecayWidth(HP::Decay::mutau, std::norm(el.mutau.second));
+
+      // Higgs to LSP decay (if model contains one)
+      s.setDecayWidth("Inv", "Inv", el.invWidth);
+
+      // all remaining partial widths
+      s.setDecayWidth("Undetected", "Undetected", el.get_undetected_width());
+   }
+
+   auto bounds = Higgs::Bounds {higgsbounds_dataset};
+   auto hbResult = bounds(pred);
+   std::vector<std::tuple<int, double, double, std::string>> hb_return {};
+   for (auto const& _hb: hbResult.selectedLimits) {
+      auto found = std::find_if(
+         std::begin(bsm_input), std::end(bsm_input),
+         [&_hb](auto const& el) { return el.particle==_hb.first; }
+      );
+      hb_return.push_back({found->pdgid, _hb.second.obsRatio(), _hb.second.expRatio(), _hb.second.limit()->to_string()});
+   }
+
+   const auto signals = Higgs::Signals {higgssignals_dataset};
+   //for (const auto &m : signals.measurements()) {
```

Comment:
Please remove this debug output or use `VERBOSE_MSG`.

Reply:
Done in 537b68cbcc9fb7bee3e3a6a53ef154f2123eca05.